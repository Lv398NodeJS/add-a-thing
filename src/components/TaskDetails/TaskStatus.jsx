import React from 'react';
import { Container, Dropdown } from 'react-bootstrap';

export default class TaskStatus extends React.Component {
  handleChangeStatus = (statusName) => {
    const { changeStatus } = this.props;
    changeStatus(false, false, statusName, false);
  };

  render() {
    const { status } = this.props;
    return (
      <Container>
        <Dropdown>
          <Dropdown.Toggle className="dropdownContainer col" size="sm" variant="primary" id="dropdown-basic">
            {status}
          </Dropdown.Toggle>
          <Dropdown.Menu>
            <Dropdown.Item onClick={() => { this.handleChangeStatus('To Do'); }}>
              {'To Do'}
            </Dropdown.Item>
            <Dropdown.Item onClick={() => { this.handleChangeStatus('In Progress'); }}>
              {'In Progress'}
            </Dropdown.Item>
            <Dropdown.Item onClick={() => { this.handleChangeStatus('Done'); }}>
              {'Done'}
            </Dropdown.Item>
          </Dropdown.Menu>
        </Dropdown>
      </Container>
    );
  }
}
