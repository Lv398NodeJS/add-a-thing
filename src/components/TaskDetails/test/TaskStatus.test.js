import React from 'react';
import { shallow } from 'enzyme';
import TaskStatus from '../TaskStatus';

describe('TaskStatus', () => {
  it('should render correctly', () => {
    const component = shallow(<TaskStatus />);
    expect(component).toMatchSnapshot();
  });
});
