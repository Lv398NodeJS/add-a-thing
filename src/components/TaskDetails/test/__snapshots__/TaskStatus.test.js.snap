// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskStatus should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TaskStatus />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "as": "div",
      "children": <ForwardRef
        as="div"
        navbar={false}
      >
        <ForwardRef(render)
          className="dropdownContainer col"
          id="dropdown-basic"
          size="sm"
          variant="primary"
        />
        <ForwardRef
          alignRight={false}
          as="div"
          flip={true}
        >
          <ForwardRef
            as={[Function]}
            disabled={false}
            onClick={[Function]}
          >
            To Do
          </ForwardRef>
          <ForwardRef
            as={[Function]}
            disabled={false}
            onClick={[Function]}
          >
            In Progress
          </ForwardRef>
          <ForwardRef
            as={[Function]}
            disabled={false}
            onClick={[Function]}
          >
            Done
          </ForwardRef>
        </ForwardRef>
      </ForwardRef>,
      "fluid": false,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "as": "div",
        "children": Array [
          <ForwardRef(render)
            className="dropdownContainer col"
            id="dropdown-basic"
            size="sm"
            variant="primary"
          />,
          <ForwardRef
            alignRight={false}
            as="div"
            flip={true}
          >
            <ForwardRef
              as={[Function]}
              disabled={false}
              onClick={[Function]}
            >
              To Do
            </ForwardRef>
            <ForwardRef
              as={[Function]}
              disabled={false}
              onClick={[Function]}
            >
              In Progress
            </ForwardRef>
            <ForwardRef
              as={[Function]}
              disabled={false}
              onClick={[Function]}
            >
              Done
            </ForwardRef>
          </ForwardRef>,
        ],
        "navbar": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": undefined,
            "className": "dropdownContainer col",
            "id": "dropdown-basic",
            "size": "sm",
            "variant": "primary",
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": undefined,
            "displayName": "Bootstrap(DropdownToggle)",
            "propTypes": undefined,
            "render": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "alignRight": false,
            "as": "div",
            "children": Array [
              <ForwardRef
                as={[Function]}
                disabled={false}
                onClick={[Function]}
              >
                To Do
              </ForwardRef>,
              <ForwardRef
                as={[Function]}
                disabled={false}
                onClick={[Function]}
              >
                In Progress
              </ForwardRef>,
              <ForwardRef
                as={[Function]}
                disabled={false}
                onClick={[Function]}
              >
                Done
              </ForwardRef>,
            ],
            "flip": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "as": [Function],
                "children": "To Do",
                "disabled": false,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "To Do",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "as": [Function],
                  "disabled": false,
                },
                "displayName": "DropdownItem",
                "render": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "as": [Function],
                "children": "In Progress",
                "disabled": false,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "In Progress",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "as": [Function],
                  "disabled": false,
                },
                "displayName": "DropdownItem",
                "render": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "as": [Function],
                "children": "Done",
                "disabled": false,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Done",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "as": [Function],
                  "disabled": false,
                },
                "displayName": "DropdownItem",
                "render": [Function],
              },
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "alignRight": false,
              "as": "div",
              "flip": true,
            },
            "displayName": "DropdownMenu",
            "render": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "Divider": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "role": "separator",
          },
          "displayName": "DropdownDivider",
          "render": [Function],
        },
        "Header": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "role": "heading",
          },
          "displayName": "DropdownHeader",
          "render": [Function],
        },
        "Item": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "as": [Function],
            "disabled": false,
          },
          "displayName": "DropdownItem",
          "render": [Function],
        },
        "Menu": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "alignRight": false,
            "as": "div",
            "flip": true,
          },
          "displayName": "DropdownMenu",
          "render": [Function],
        },
        "Toggle": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "displayName": "Bootstrap(DropdownToggle)",
          "propTypes": undefined,
          "render": [Function],
        },
        "defaultProps": Object {
          "as": "div",
          "navbar": false,
        },
        "displayName": "Dropdown",
        "render": [Function],
      },
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "defaultProps": Object {
        "as": "div",
        "fluid": false,
      },
      "displayName": "Container",
      "render": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "as": "div",
        "children": <ForwardRef
          as="div"
          navbar={false}
        >
          <ForwardRef(render)
            className="dropdownContainer col"
            id="dropdown-basic"
            size="sm"
            variant="primary"
          />
          <ForwardRef
            alignRight={false}
            as="div"
            flip={true}
          >
            <ForwardRef
              as={[Function]}
              disabled={false}
              onClick={[Function]}
            >
              To Do
            </ForwardRef>
            <ForwardRef
              as={[Function]}
              disabled={false}
              onClick={[Function]}
            >
              In Progress
            </ForwardRef>
            <ForwardRef
              as={[Function]}
              disabled={false}
              onClick={[Function]}
            >
              Done
            </ForwardRef>
          </ForwardRef>
        </ForwardRef>,
        "fluid": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "div",
          "children": Array [
            <ForwardRef(render)
              className="dropdownContainer col"
              id="dropdown-basic"
              size="sm"
              variant="primary"
            />,
            <ForwardRef
              alignRight={false}
              as="div"
              flip={true}
            >
              <ForwardRef
                as={[Function]}
                disabled={false}
                onClick={[Function]}
              >
                To Do
              </ForwardRef>
              <ForwardRef
                as={[Function]}
                disabled={false}
                onClick={[Function]}
              >
                In Progress
              </ForwardRef>
              <ForwardRef
                as={[Function]}
                disabled={false}
                onClick={[Function]}
              >
                Done
              </ForwardRef>
            </ForwardRef>,
          ],
          "navbar": false,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": undefined,
              "className": "dropdownContainer col",
              "id": "dropdown-basic",
              "size": "sm",
              "variant": "primary",
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": undefined,
              "displayName": "Bootstrap(DropdownToggle)",
              "propTypes": undefined,
              "render": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "alignRight": false,
              "as": "div",
              "children": Array [
                <ForwardRef
                  as={[Function]}
                  disabled={false}
                  onClick={[Function]}
                >
                  To Do
                </ForwardRef>,
                <ForwardRef
                  as={[Function]}
                  disabled={false}
                  onClick={[Function]}
                >
                  In Progress
                </ForwardRef>,
                <ForwardRef
                  as={[Function]}
                  disabled={false}
                  onClick={[Function]}
                >
                  Done
                </ForwardRef>,
              ],
              "flip": true,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "as": [Function],
                  "children": "To Do",
                  "disabled": false,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "To Do",
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "defaultProps": Object {
                    "as": [Function],
                    "disabled": false,
                  },
                  "displayName": "DropdownItem",
                  "render": [Function],
                },
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "as": [Function],
                  "children": "In Progress",
                  "disabled": false,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "In Progress",
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "defaultProps": Object {
                    "as": [Function],
                    "disabled": false,
                  },
                  "displayName": "DropdownItem",
                  "render": [Function],
                },
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "as": [Function],
                  "children": "Done",
                  "disabled": false,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Done",
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "defaultProps": Object {
                    "as": [Function],
                    "disabled": false,
                  },
                  "displayName": "DropdownItem",
                  "render": [Function],
                },
              },
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "alignRight": false,
                "as": "div",
                "flip": true,
              },
              "displayName": "DropdownMenu",
              "render": [Function],
            },
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "Divider": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "role": "separator",
            },
            "displayName": "DropdownDivider",
            "render": [Function],
          },
          "Header": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "role": "heading",
            },
            "displayName": "DropdownHeader",
            "render": [Function],
          },
          "Item": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "as": [Function],
              "disabled": false,
            },
            "displayName": "DropdownItem",
            "render": [Function],
          },
          "Menu": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "alignRight": false,
              "as": "div",
              "flip": true,
            },
            "displayName": "DropdownMenu",
            "render": [Function],
          },
          "Toggle": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": undefined,
            "displayName": "Bootstrap(DropdownToggle)",
            "propTypes": undefined,
            "render": [Function],
          },
          "defaultProps": Object {
            "as": "div",
            "navbar": false,
          },
          "displayName": "Dropdown",
          "render": [Function],
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": Object {
          "as": "div",
          "fluid": false,
        },
        "displayName": "Container",
        "render": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
